### PR TITLE
Port to using multipackage for CI scripts

### PR DESCRIPTION
**Pull request by timburke**
This PR is the result of installing `multpackage` in this repo to manage the CI process.  It also removes the old `scripts` folder that is obsoleted by multipackage.

### CODE REVIEW

File: .multipackage/scripts/generate_api.py [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+def delete_with_retry(folder):
+    """Try multiple times to delete a folder.
+
+    This is required on windows because of things like:
+    https://bugs.python.org/issue15496
+    https://blogs.msdn.microsoft.com/oldnewthing/20120907-00/?p=6663/
+    https://mail.python.org/pipermail/python-dev/2013-September/128350.html
+    """
+
+    for _i in range(0, 5):
```
Comment (mattrunchey) [inline .multipackage/scripts/generate_api.py]: technically you can run this loop with just `for _ in range(0,5)`

Comment (timburke) [reply .multipackage/scripts/generate_api.py]: Good catch, fixed in `multipackage`, will get updated the next time we update the build scripts in this repository.

File: .multipackage/scripts/generate_api.py [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+def verify_output(path):
+    """Verify that output path exists and is a directory.
+
+    Args:
+        path (str): The destination path.
+    """
+
+    if not os.path.exists(path):
+        os.mkdir(path)
```
Comment (mattrunchey) [inline .multipackage/scripts/generate_api.py]: If the user puts in a path with nested non-created directories, I think mkdir will fail:
```
>>> os.mkdir("test/test2/test3")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
OSError: [Errno 2] No such file or directory: 'test/test2/test3'
>>> os.getcwd()
'/Users/mrunchey/gitrepos/arch_gateway'
>>> os.mkdir("test/test2")
>>> os.mkdir("test/test2/test3")
```

File: .multipackage/scripts/release_component.py [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+import requests
+import setuptools.sandbox
+from twine.commands.upload import upload
+from twine.settings import Settings
+
+if sys.version_info.major < 3:
+    from builtins import raw_input as input
+
+try:
+    from shared_errors import MismatchError, ExternalError, InternalError, handle_exception  #pylint:disable=relative-import;We need this logic so that we work when installed
```
[the remark this answers is not in the document]
Comment (timburke) [reply .multipackage/scripts/release_component.py]: During unit tests on the `multipackage` package, this script will be invoked from inside the `multipackage` package so it needs a relative import on python 3 to access `shared_errors.py` in the same subpackage.  However, that entire subpackage is copied into `.multipackage/scripts` as standalone scripts where relative imports won't work, so it needs to be able to do the import in both relative and absolute so it works both during testing and when its installed:

https://github.com/iotile/multipackage/tree/master/multipackage/data/scripts

File: .multipackage/scripts/generate_api.py (hunk 2 above)
Comment (timburke) [reply .multipackage/scripts/generate_api.py]: Agreed, normally I would use `os.makedirs` for that reason but before python 3 it didn't (apparently) support if the directory already existed:

https://docs.python.org/3/library/os.html#os.makedirs

I created https://github.com/iotile/multipackage/issues/11 to track this.

**Pull request merged**